src/Orleans/Statistics/ThreadTrackingStatistic.cs

```diff
@@ -241,5 +242,40 @@ private void TrackContextSwitches()
             }
 #endif
         }
+
+        #region IDisposable Support
+        private bool disposedValue = false; // To detect redundant calls
+
+        protected virtual void Dispose(bool disposing)
```

Comment:
Same comment here - consider suppressing the warning instead of adding no-op code. Curious what others think.